sdk/storage/azure-storage-blob/azure/storage/blob/_shared/policies.py

```diff
@@ -132,20 +124,31 @@ def on_request(self, request: "PipelineRequest") -> None:
         custom_id = request.context.options.pop('client_request_id', None)
         request.http_request.headers['x-ms-client-request-id'] = custom_id or str(uuid.uuid1())
 
-    # def on_response(self, request, response):
-    #     # raise exception if the echoed client request id from the service is not identical to the one we sent
-    #     if self.request_id_header_name in response.http_response.headers:
-
-    #         client_request_id = request.http_request.headers.get(self.request_id_header_name)
-
-    #         if response.http_response.headers[self.request_id_header_name] != client_request_id:
-    #             raise AzureError(
-    #                 "Echoed client request ID: {} does not match sent client request ID: {}.  "
-    #                 "Service request ID: {}".format(
-    #                     response.http_response.headers[self.request_id_header_name], client_request_id,
-    #                     response.http_response.headers['x-ms-request-id']),
-    #                 response=response.http_response
-    #             )
+    def on_response(self, request: "PipelineRequest", response: "PipelineResponse") -> None:
+        is_success = response.http_response.status_code < 300
+        # Validate structured body PUT/GET requests
+        if ('x-ms-structured-body' in request.http_request.headers and
+                'x-ms-structured-content-length' in request.http_request.headers and
+                'x-ms-structured-body' not in response.http_response.headers and is_success):
+            raise ValueError("Response did not acknowledge structured body. "
+                             "Unexpected data may have been persisted to storage.")
+        elif ('x-ms-structured-body' in request.http_request.headers and
+              'x-ms-structured-body' not in response.http_response.headers and is_success):
+            raise ValueError("Response did not acknowledge structured body. Unknown structure in response body.")
+
+    # # raise exception if the echoed client request id from the service is not identical to the one we sent
+    # if self.request_id_header_name in response.http_response.headers:
+    #
+    #     client_request_id = request.http_request.headers.get(self.request_id_header_name)
+    #
+    #     if response.http_response.headers[self.request_id_header_name] != client_request_id:
+    #         raise AzureError(
+    #             "Echoed client request ID: {} does not match sent client request ID: {}.  "
+    #             "Service request ID: {}".format(
+    #                 response.http_response.headers[self.request_id_header_name], client_request_id,
+    #                 response.http_response.headers['x-ms-request-id']),
+    #             response=response.http_response
+    #         )
```

Comment:
nit: Is this intended to be committed as a comment?

Reply:
Yes, it was there as a comment before so I left :)